pkg/resources/digitalocean/cloud.go

```diff
@@ -278,3 +287,172 @@ func (c *Cloud) getEtcdClusterSpec(volumeName string, dropletName string) (*etcd
+			})
+
+		} else {
+			// This is a node (add this instance group only once)
+			nodeExists = true
+			workerDroplets = append(workerDroplets, strconv.Itoa(droplet.ID))
+		}
+	}
+
+	if nodeExists {
+		instanceGroupName = fmt.Sprintf("%s-%s", clusterName, "nodes")
+
+		result = append(result, DOInstanceGroup{
+			InstanceGroupName: instanceGroupName,
+			GroupType:         "node",
+			ClusterName:       clusterName,
+			Members:           workerDroplets,
+		})
+	}
+
+	klog.V(8).Infof("InstanceGroup Info = %v", result)
+
+	return result, nil
+}
+
+func getDropletInstanceGroup(tags []string) (string, error) {
+	for _, tag := range tags {
+		klog.V(8).Infof("Check tag = %s", tag)
+		if strings.Contains(strings.ToLower(tag), TagKubernetesInstanceGroup) {
+			tagParts := strings.Split(tag, ":")
+			if len(tagParts) < 2 {
+				return "", fmt.Errorf("tag split failed, too few components for tag %q", tag)
+			}
+			return tagParts[1], nil
+		}
+	}
+
+	return "", fmt.Errorf("Didn't find k8s-index for tag %v", tags)
+}
+
+func getDropletIndexFromTag(tags []string) (string, error) {
+	for _, tag := range tags {
+		klog.V(8).Infof("Check tag = %s", tag)
+		if strings.Contains(strings.ToLower(tag), TagKubernetesClusterIndex) {
+			tagParts := strings.Split(tag, ":")
+			if len(tagParts) < 2 {
+				return "", fmt.Errorf("tag split failed, too few components for tag %q", tag)
+			}
+			return tagParts[1], nil
+		}
+	}
+
+	return "", fmt.Errorf("Didn't find k8s-index for tag %v", tags)
+}
+
+// matchInstanceGroup filters a list of instancegroups for recognized cloud groups
+func matchInstanceGroup(name string, clusterName string, instancegroups []*kops.InstanceGroup) (*kops.InstanceGroup, error) {
+	var instancegroup *kops.InstanceGroup
+	for _, g := range instancegroups {
+		var groupName string
+
+		switch g.Spec.Role {
+		case kops.InstanceGroupRoleMaster, kops.InstanceGroupRoleNode:
+			groupName = clusterName + "-" + g.ObjectMeta.Name
+		default:
+			klog.Warningf("Ignoring InstanceGroup of unknown role %q", g.Spec.Role)
+			continue
+		}
+
+		if name == groupName {
+			if instancegroup != nil {
+				return nil, fmt.Errorf("found multiple instance groups matching servergrp %q", groupName)
+			}
+			instancegroup = g
+		}
+	}
+
+	return instancegroup, nil
+}
+
+func buildCloudInstanceGroup(c *Cloud, ig *kops.InstanceGroup, g DOInstanceGroup, nodeMap map[string]*v1.Node) (*cloudinstances.CloudInstanceGroup, error) {
+	cg := &cloudinstances.CloudInstanceGroup{
+		HumanName:     g.InstanceGroupName,
+		InstanceGroup: ig,
+		Raw:           g,
+		MinSize:       int(fi.Int32Value(ig.Spec.MinSize)),
+		TargetSize:    int(fi.Int32Value(ig.Spec.MinSize)),
+		MaxSize:       int(fi.Int32Value(ig.Spec.MaxSize)),
+	}
+
+	for _, member := range g.Members {
+
+		// DO doesn't have a notion of Auto Scaling Group - use the same config Type for both current config and new config.
+		err := cg.NewCloudInstanceGroupMember(member, g.GroupType, g.GroupType, nodeMap)
```

Comment:
This means DO won't be able to know when to do a rolling update. All rolling updates would  have to use `--force`.

Reply:
Looking at the [DO Cloud](https://github.com/kubernetes/kops/tree/master/upup/pkg/fi/cloudup/do), it doesn't (yet) appear to support deleting instances, so I'm guessing rolling updates don't work at all. Perhaps @srikiz can confirm?

Reply:
yes @rifelpet - that is something I want to check next.

Reply:
Yeah, looks like DO doesn't support rolling updates. So that's not a blocker for this PR.

The comment is a bit off-point—the second and third parameters just need to be unequal whenever any field of the `godo.DropletCreateRequest` used to create the instance was different than what would be used to create a new instance. This can be implemented as either a generation id or a hash of the fields.

Reply:
sure, thanks @johngmyers  - I'll keep a note of this when I implement rolling update feature.